src/uct/base/uct_iface.h

```diff
@@ -449,12 +449,26 @@ uct_iface_invoke_am(uct_base_iface_t *iface, uint8_t id, void *data,
  * @param data   Optional completion data (operation reply).
  */
 static UCS_F_ALWAYS_INLINE
-void uct_invoke_completion(uct_completion_t *comp)
+void uct_invoke_completion(uct_completion_t *comp, ucs_status_t status)
 {
-    ucs_trace_func("comp=%p, count=%d", comp, comp->count);
+    ucs_trace_func("comp=%p, count=%d, status=%d", comp, comp->count, status);
     if (--comp->count == 0) {
-        comp->func(comp);
+        comp->func(comp, status);
     }
 }
 
+
+/**
+ * Set error on interface.
+ *
+ * In the event of an error rendering the interface temporarily or permanently
+ * disfunctional, this function causes further calls to the interface to return
+ * the given error code - except for a call to iface_destroy().
+ *
+ * @param iface  Subject interface.
+ * @param status Error code to set.
+ */
+void uct_base_iface_set_error(uct_iface_h iface, ucs_status_t status);
```

Comment:
The function is to be used wherever an "persistent" error occurs - and subsequent calls must return the same error. I didn't put in the calls so that the build may pass. This will be done in the next phase of development.